### ISSUE TITLE
how to keep the last record when add an unique index to a table?

### ISSUE THREAD

**Issue by gongfangjun** [github/gh-ost#423]
Hi all, when I try to add an unique index to a table, I found out that:

        if two records are same, gh-ost will always keep the first one

And in general, we want to keep the last one, so, is there a way 
to keep the last one?

**Comment by shlomi-noach** [comment]
Hi @gongfangjun ,

Where did you get this quote from?

>  if two records are same, gh-ost will always keep the first one

**Comment by gongfangjun** [comment]
@shlomi-noach 
  hi, I did a little test and found this, let's have a look at the data in the two tables:

mysql> select * from user_num;
+----+--------+-----+-----+
| id | uname  | num | age |
+----+--------+-----+-----+
|  1 | maria  | 320 |  22 |
|  2 | olivia | 355 |  18 |
|  4 | zuna   | 872 |  19 |
+----+--------+-----+-----+
3 rows in set (0.00 sec)

mysql> select * from _user_num_del;
+----+--------+-----+-----+
| id | uname  | num | age |
+----+--------+-----+-----+
|  1 | maria  | 320 |  22 |
|  2 | olivia | 355 |  18 |
|  3 | maria  | 320 |  22 |
|  4 | zuna   | 872 |  19 |
+----+--------+-----+-----+
4 rows in set (0.00 sec)

the value of gh-ost  parameter "-alter" is : "add unique name_num(uname,num)"
after the job done,I noticed that ,the data which id = 3 was disappeared,
that is how I get this conclusion

**Comment by shlomi-noach** [comment]
Oh ok, thank you, it seemed like a quote and I searched all over for the source.

The official answer is that the choice of rows in the case of adding a `unique key` is _undefined_.

So, the real answer is that the choice of row really depends on your workload. In your case, this was a static table, and the nature of `insert ignore into... select...` is to use the first, discard the second. However at some time in the future `gh-ost` will not be using a `insert into... select` and thus the behavior will change.

If any writes are coming through, then real chaos begins. It would depend which row was updated, when that took place in the duration of the copy... 

So it's really undefined and unfortunately I will not make anything to necessarily keep the last one.
